File: Backend/src/utils/templates/python/linked_list.py
from typing import *
import sys, json, traceback

# ###USERCODE###

class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

def __build_list(vals):
    if not vals:
        return None
    dummy = ListNode(0)
    cur = dummy
    for v in vals:
        cur.next = ListNode(v)
        cur = cur.next
    return dummy.next

def __serialize_list(head):
    result = []
    seen = set()
    while head:
        if id(head) in seen:
            result.append("cycle")
            break
        seen.add(id(head))
        result.append(str(head.val))
        head = head.next
    return "[" + ",".join(result) + "]"

def __format(result):
    if result is None:
        return "null"
    if isinstance(result, bool):
        return "true" if result else "false"
    if isinstance(result, ListNode):
        return __serialize_list(result)
    if isinstance(result, list):
        return json.dumps(result)
    if isinstance(result, float):
        return f"{result:.5f}"
    return str(result)

if __name__ == '__main__':
    raw = sys.stdin.read().strip()
    lines = [l.strip() for l in raw.splitlines() if l.strip()]
    args = []
    for line in lines:
        try:
            val = json.loads(line)
        except Exception:
            val = line
        # Convert list args that represent linked lists
        if isinstance(val, list) and all(isinstance(x, int) for x in val):
            args.append(__build_list(val))
        else:
            args.append(val)
    try:
        obj = Solution()
        # Entry point is replaced at runtime by code-runner.ts
        result = getattr(obj, '###ENTRYPOINT###')(*args)
        print(__format(result))
    except Exception:
        sys.stderr.write(traceback.format_exc())
        sys.exit(1)

File: Backend/src/utils/templates/javascript/linked_list.js
// ###USERCODE###

class ListNode {
    constructor(val = 0, next = null) {
        this.val = val; this.next = next;
    }
}

function __buildList(vals) {
    if (!vals || vals.length === 0) return null;
    const dummy = new ListNode(0);
    let cur = dummy;
    for (const v of vals) { cur.next = new ListNode(v); cur = cur.next; }
    return dummy.next;
}

function __serializeList(head) {
    const result = [];
    const seen = new Set();
    while (head) {
        if (seen.has(head)) { result.push('cycle'); break; }
        seen.add(head);
        result.push(String(head.val));
        head = head.next;
    }
    return '[' + result.join(',') + ']';
}

function __format(result) {
    if (result === null || result === undefined) return 'null';
    if (typeof result === 'boolean') return result ? 'true' : 'false';
    if (result instanceof ListNode) return __serializeList(result);
    if (Array.isArray(result)) return JSON.stringify(result);
    return String(result);
}

(function __main() {
    const lines = require('fs').readFileSync(0, 'utf-8').trim().split('\n').filter(Boolean);
    const args = lines.map(line => {
        try {
            const val = JSON.parse(line);
            if (Array.isArray(val) && val.every(x => typeof x === 'number')) return __buildList(val);
            return val;
        } catch (e) { return line; }
    });
    try {
        const obj = new Solution();
        const result = obj['###ENTRYPOINT###'](...args);
        process.stdout.write(__format(result) + '\n');
    } catch (e) {
        process.stderr.write((e.stack || String(e)) + '\n');
        process.exit(1);
    }
})();
